src/Sylius/Bundle/CoreBundle/Modifier/OrderItemQuantityModifier.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Modifier;
+
+use Sylius\Bundle\OrderBundle\Factory\OrderItemUnitFactoryInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class OrderItemQuantityModifier implements OrderItemQuantityModifierInterface
+{
+    /**
+     * @var OrderItemUnitFactoryInterface
+     */
+    private $orderItemUnitFactory;
+
+    /**
+     * @param OrderItemUnitFactoryInterface $orderItemUnitFactory
+     */
+    public function __construct(OrderItemUnitFactoryInterface $orderItemUnitFactory)
+    {
+        $this->orderItemUnitFactory = $orderItemUnitFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function modify(OrderItemInterface $orderItem, $targetQuantity)
+    {
+        $itemQuantity = $orderItem->getQuantity();
+
+        if ($targetQuantity > $itemQuantity) {
+            $this->increaseUnitsNumber($orderItem, $targetQuantity - $itemQuantity);
+        } else if ($targetQuantity < $itemQuantity) {
```

Comment:
Well, ``else`` is part of ``else if``, so I think it's lesser evil but still evil :smile: :smiling_imp: However acceptable in this case